nsxt/resource_nsxt_vpc_nat_rule_test.go

```diff
@@ -0,0 +1,308 @@
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceNsxtVpcNatRule_importBasic(t *testing.T) {
+	name := getAccTestResourceName()
+	testResourceName := "nsxt_vpc_nat_rule.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNsxtVpcNatRuleCheckDestroy(state, name)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNsxtVpcNatRuleMinimalistic(),
+			},
+			{
+				ResourceName:      testResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccResourceNsxtPolicyImportIDRetriever(testResourceName),
+			},
+		},
+	})
+}
+
+func testAccNsxtVpcNatRuleExists(displayName string, resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+
+		connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Policy VpcNatRule resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("Policy VpcNatRule resource ID not set in resources")
+		}
+
+		parentPath := rs.Primary.Attributes["parent_path"]
+
+		exists, err := resourceNsxtPolicyVpcNatRuleExists(testAccGetSessionContext(), parentPath, resourceID, connector)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return fmt.Errorf("VpcNatRule %s does not exist", resourceID)
+		}
+
+		return nil
+	}
+}
+
+func testAccNsxtVpcNatRuleCheckDestroy(state *terraform.State, displayName string) error {
+	connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_vpc_nat_rule" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		parentPath := rs.Primary.Attributes["parent_path"]
+
+		exists, err := resourceNsxtPolicyVpcNatRuleExists(testAccGetSessionContext(), parentPath, resourceID, connector)
+		if err == nil {
+			return err
+		}
+
+		if exists {
+			return fmt.Errorf("VpcNatRule %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNsxtVpcNatRulePrerequisites() string {
+	return fmt.Sprintf(`
+data "nsxt_vpc_nat" "test" {
+  %s
+  nat_type = "USER"
+}
+
+resource "nsxt_vpc_ip_address_allocation" "test" {
+  %s
+  display_name = "%s"
+}
+`, testAccNsxtPolicyMultitenancyContext(), testAccNsxtPolicyMultitenancyContext(), accTestVpcNatIPAllocationTestName)
+}
+
+func testAccNsxtVpcNatRuleTemplate(createFlow bool) string {
+	var attrMap map[string]string
+	if createFlow {
+		attrMap = accTestVpcNatRuleCreateAttributes
+	} else {
+		attrMap = accTestVpcNatRuleUpdateAttributes
+	}
+	return testAccNsxtVpcNatRulePrerequisites() + fmt.Sprintf(`
+resource "nsxt_vpc_nat_rule" "test" {
+  parent_path         = data.nsxt_vpc_nat.test.path
```

Comment:
Do we have this DS?

Reply:
Yes, sorry it was missing in original PR, its here now :)